File: frontend/src/notifications/services/notificationService.ts
import notificationApi from '../api/notificationApi';
import { NotificationType } from '../types';
import type { NotificationService as LocalNotificationService } from '../hooks/useNotificationService';

/**
 * 通知服务类
 * 处理通知相关的业务逻辑
 */
export class NotificationService {
  /**
   * 获取用户通知列表
   * @param params 查询参数
   * @returns 通知列表数据
   */
  static async getNotifications(params?: {
    page?: number;
    size?: number;
    isRead?: boolean;
    type?: string;
  }) {
    try {
      const response = await notificationApi.getNotifications(params);
      if (response.success && response.data) {
        return response.data;
      }
      throw new Error(response.message || '获取通知列表失败');
    } catch (error) {
      console.error('获取通知列表失败:', error);
      throw error;
    }
  }

  /**
   * 获取未读通知数量
   * @returns 未读通知数量
   */
  static async getUnreadCount() {
    try {
      const response = await notificationApi.getUnreadCount();
      if (response.success && response.data !== undefined) {
        return response.data;
      }
      throw new Error(response.message || '获取未读通知数量失败');
    } catch (error) {
      console.error('获取未读通知数量失败:', error);
      throw error;
    }
  }

  /**
   * 标记单个通知为已读
   * @param notificationId 通知ID
   */
  static async markAsRead(notificationId: string) {
    try {
      const response = await notificationApi.markAsRead(notificationId);
      if (!response.success) {
        throw new Error(response.message || '标记通知已读失败');
      }
    } catch (error) {
      console.error('标记通知已读失败:', error);
      throw error;
    }
  }

  /**
   * 标记所有通知为已读
   */
  static async markAllAsRead() {
    try {
      const response = await notificationApi.markAllAsRead();
      if (!response.success) {
        throw new Error(response.message || '标记所有通知已读失败');
      }
    } catch (error) {
      console.error('标记所有通知已读失败:', error);
      throw error;
    }
  }

  /**
   * 删除单个通知
   * @param notificationId 通知ID
   */
  static async deleteNotification(notificationId: string) {
    try {
      const response = await notificationApi.deleteNotification(notificationId);
      if (!response.success) {
        throw new Error(response.message || '删除通知失败');
      }
    } catch (error) {
      console.error('删除通知失败:', error);
      throw error;
    }
  }

  /**
   * 批量删除通知
   * @param notificationIds 通知ID数组
   */
  static async deleteNotifications(notificationIds: string[]) {
    try {
      const response = await notificationApi.deleteNotifications(notificationIds);
      if (!response.success) {
        throw new Error(response.message || '批量删除通知失败');
      }
    } catch (error) {
      console.error('批量删除通知失败:', error);
      throw error;
    }
  }

  /**
   * 显示成功通知
   * @param notificationService local notification service实例
   * @param message 通知消息
   * @param title 通知标题
   */
  static showSuccess(
    notificationService: LocalNotificationService,
    message: string,
    title?: string
  ) {
    notificationService.showSuccess(message, title ? { title } : {});
  }

  /**
   * 显示错误通知
   * @param notificationService local notification service实例
   * @param message 通知消息
   * @param title 通知标题
   */
  static showError(
    notificationService: LocalNotificationService,
    message: string,
    title?: string
  ) {
    notificationService.showError(message, title ? { title } : {});
  }

  /**
   * 显示警告通知
   * @param notificationService local notification service实例
   * @param message 通知消息
   * @param title 通知标题
   */
  static showWarning(
    notificationService: LocalNotificationService,
    message: string,
    title?: string
  ) {
    notificationService.showWarning(message, title ? { title } : {});
  }

  /**
   * 显示信息通知
   * @param notificationService local notification service实例
   * @param message 通知消息
   * @param title 通知标题
   */
  static showInfo(
    notificationService: LocalNotificationService,
    message: string,
    title?: string
  ) {
    notificationService.showInfo(message, title ? { title } : {});
  }

  /**
   * 显示自定义类型的通知
   * @param notificationService local notification service实例
   * @param message 通知消息
   * @param options 通知选项
   */
  static showNotification(
    notificationService: LocalNotificationService,
    message: string,
    options?: {
      title?: string;
      type?: NotificationType;
      duration?: number;
      onClick?: () => void;
    }
  ) {
    notificationService.showNotification(message, options);
  }
}

export default NotificationService;

File: backend/src/main/java/org/backend/service/NotificationService.java
package org.backend.service;

import org.backend.A_general.base.service.BaseService;
import org.backend.dto.response.BaseResponse;
import org.backend.entity.Notification;
import org.backend.entity.enums.NotificationType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * 通知服务接口
 * 提供通知管理相关的业务逻辑
 */
public interface NotificationService extends BaseService<Notification, Long> {

    /**
     * 获取用户的通知列表
     * @param userId 用户ID
     * @param pageable 分页参数
     * @return 通知分页列表
     */
    Page<Notification> getNotificationsByUserId(Long userId, Pageable pageable);

    /**
     * 根据条件筛选通知列表
     * @param userId 用户ID
     * @param isRead 是否已读
     * @param type 通知类型
     * @param pageable 分页参数
     * @return 通知分页列表
     */
    Page<Notification> getNotificationsByCondition(Long userId, Boolean isRead, NotificationType type, Pageable pageable);

    /**
     * 获取用户未读通知数量
     * @param userId 用户ID
     * @return 未读通知数量
     */
    long getUnreadCount(Long userId);

    /**
     * 标记单个通知为已读
     * @param notificationId 通知ID
     * @param userId 用户ID
     * @return 操作结果
     */
    boolean markAsRead(Long notificationId, Long userId);

    /**
     * 标记所有通知为已读
     * @param userId 用户ID
     * @return 操作结果
     */
    boolean markAllAsRead(Long userId);

    /**
     * 删除单个通知
     * @param notificationId 通知ID
     * @param userId 用户ID
     * @return 操作结果
     */
    boolean deleteNotification(Long notificationId, Long userId);

    /**
     * 批量删除通知
     * @param notificationIds 通知ID列表
     * @param userId 用户ID
     * @return 操作结果
     */
    boolean deleteNotifications(List<Long> notificationIds, Long userId);

    /**
     * 创建并发送通知
     * @param userId 用户ID
     * @param title 标题
     * @param message 内容
     * @param type 类型
     * @return 创建的通知
     */
    Notification createNotification(Long userId, String title, String message, NotificationType type);

    /**
     * 创建并发送通知（带额外参数）
     * @param userId 用户ID
     * @param title 标题
     * @param message 内容
     * @param type 类型
     * @param duration 显示时长
     * @param targetUrl 点击跳转URL
     * @return 创建的通知
     */
    Notification createNotification(Long userId, String title, String message, NotificationType type, Integer duration, String targetUrl);
}